docs/developers/developer-reference/smart-contract-developer-best-practices.md

```diff
@@ -0,0 +1,274 @@
+---
+id: smart-contract-developer-best-practices
+title: Smart Contract Developer Best Practices
+---
+
+## Code Arrangement
+
+We'll start with something simple: code arrangement. It's best to separate your code into 4 main parts:
+- endpoints
+- view functions
+- private functions
+- storage
+
+This ensures that your it's much easier to find what you're looking for, and it's also much easier for everyone else who's working on that smart contract. Additionally, it's also best to split endpoints by their level of access. Some endpoints might be owner-only, some might be usable only by a select few addresses from a whitelist, and some can be called by anyone.  
+
+The recommended order is the one from the list above, but order is not important as long as you clearly separate your code. Even better if you split those into modules.  
+
+## BigUint Operations
+
+`BigUint` is simply a handle to the actual representation, similar to how system file handles work, so it's simply a struct with an `i32` as member, representing the handle. All the operations have to be done through API functions, passing the handles for result, first operand, second operand. Using Rust's operator overloading feature, we're able to overwrite arithmetic operators and provide an easy way of adding `BigUint`s, just like primitive number types.  
+
+Even so, you might easily reach situations where you want to use the same number multiple times. For instance, let's say you have 4 `BigUint`s: A, B, C, D, and you want to do the following operations:
```

Comment:
I think it would be better to have the variables in lowercase, since uppercase single letters are usually either generics or constants.
You can highlight them as code using backticks, so:

```
`BigUint`s: `a`, `b`, `c`, `d`, and you want to do the following operations:
```

Reply:
Done.